core/src/vault_registry.rs

```diff
@@ -0,0 +1,433 @@
+use std::mem::size_of;
+
+use bytemuck::{Pod, Zeroable};
+use jito_bytemuck::{
+    types::{PodU128, PodU64},
+    AccountDeserialize, Discriminator,
+};
+use shank::{ShankAccount, ShankType};
+use solana_program::{account_info::AccountInfo, msg, program_error::ProgramError, pubkey::Pubkey};
+
+use crate::{
+    constants::{MAX_ST_MINTS, MAX_VAULT_OPERATOR_DELEGATIONS},
+    discriminators::Discriminators,
+    error::TipRouterError,
+    ncn_fee_group::NcnFeeGroup,
+};
+
+#[derive(Debug, Clone, Copy, Zeroable, ShankType, Pod)]
+#[repr(C)]
+pub struct StMintEntry {
+    st_mint: Pubkey,
+    ncn_fee_group: NcnFeeGroup,
+    reward_multiplier_bps: PodU64,
+    switchboard_feed: Pubkey,
+    no_feed_weight: PodU128,
+    reserved: [u8; 32],
```

Comment:
would add more reserve space here, could see that being useful for other feed ids or metadata